include/unifex/transform_done.hpp

```diff
@@ -0,0 +1,364 @@
+      -> std::invoke_result_t<CPO, const Receiver&, Args...> {
+    return std::move(cpo)(r.get_receiver(), (Args&&)args...);
+  }
+  
+  template <typename VisitFunc>
+  friend void tag_invoke(
+      tag_t<visit_continuations>,
+      const type& r,
+      VisitFunc&& func) noexcept(std::is_nothrow_invocable_v<
+                                VisitFunc&,
+                                const Receiver&>) {
+    std::invoke(func, r.get_receiver());
+  }
+
+  const Receiver& get_receiver() const noexcept {
+    assert(op_ != nullptr);   
+    return op_->receiver_;
+  }
+
+  operation* op_;
+};
+
+template<typename Source, typename Done, typename Receiver>
+class _frcvr<Source, Done, Receiver>::type {
+  using operation = operation_type<Source, Done, Receiver>;
+
+public:
+  explicit type(operation* op) noexcept
+  : op_(op) {}
+
+  type(type&& other) noexcept
+  : op_(std::exchange(other.op_, {}))
+  {}
+ 
+  template<
+    typename R = Receiver,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_value), R>, int> = 0>
+  void set_value() noexcept(std::is_nothrow_invocable_v<tag_t<unifex::set_value>&, Receiver>) {
+    assert(op_ != nullptr);
+    unifex::set_value(std::move(op_->receiver_));
+  }
+
+  template<
+    typename R = Receiver,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_done), R>, int> = 0>
+  void set_done() noexcept {
+    assert(op_ != nullptr);
+    unifex::set_done(std::move(op_->receiver_));
+  }
+
+  template<
+    typename Error,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_error), Receiver, Error>, int> = 0>
+  void set_error(Error&& error) noexcept {
+    assert(op_ != nullptr);
+    unifex::set_error(std::move(op_->receiver_), (Error&&)error);
+  }
+
+private:
+  template<typename CPO, typename... Args>
+  friend auto tag_invoke(CPO cpo, const type& r, Args&&... args)
+      noexcept(std::is_nothrow_invocable_v<CPO, const Receiver&, Args...>)
+      -> std::invoke_result_t<CPO, const Receiver&, Args...> {
+    return std::move(cpo)(r.get_receiver(), (Args&&)args...);
+  }
+  
+  template <typename VisitFunc>
+  friend void tag_invoke(
+      tag_t<visit_continuations>,
+      const type& r,
+      VisitFunc&& func) noexcept(std::is_nothrow_invocable_v<
+                                VisitFunc&,
+                                const Receiver&>) {
+    std::invoke(func, r.get_receiver());
+  }
+
+  const Receiver& get_receiver() const noexcept {
+    assert(op_ != nullptr);   
+    return op_->receiver_;
+  }
+
+  operation* op_;
+};
+
+template<typename Source, typename Done, typename Receiver>
+class _op<Source, Done, Receiver>::type {
+  using source_receiver = receiver_type<Source, Done, Receiver>;
+  using final_receiver = final_receiver_type<Source, Done, Receiver>;
+
+public:
+  template<typename Source2, typename Done2, typename Receiver2>
+  explicit type(Source2&& source, Done2&& done, Receiver2&& dest)
```

Comment:
As we are immediately calling connect() on the `source` object we shouldn't need to have an extra `Source2` template arg here. We should be just always passing a `Source&&` at the call-site.